iceberg/iceberg-catalog/src/main/java/org/apache/iceberg/hive/HiveSchemaUtil.java

```diff
@@ -134,6 +136,28 @@ public static Type convert(TypeInfo typeInfo) {
     return HiveSchemaConverter.convert(typeInfo, false);
   }
 
+  /**
+   * Produces the difference of two FieldSchema lists by only taking into account the field name and type.
+   * @param from List of fields to subtract from
+   * @param to List of fields to subtract
+   * @return the result list of difference
+   */
+  public static List<FieldSchema> schemaDifference(List<FieldSchema> from, List<FieldSchema> to) {
+    List<FieldSchema> result = new LinkedList<>(from);
```

Comment:
Why use `List` as an input? `Collection` might serve as well.
Why not use lambdas?
Maybe just create a name/type map from the `to` list and just filter the results with a lambda?